pkg/controller/admissionchecks/provisioning/controller_test.go

```diff
@@ -1122,25 +1153,58 @@ func TestReconcile(t *testing.T) {
 					Reason:    "ProvisioningRequestCreated",
 					Message:   `Created ProvisioningRequest: "wl-check1-2"`,
 				},
+			}},
+		"workload retries the admission check when is not admitted and receives the provisioning request's BookingExpired condition": {
+			workload: (&utiltesting.WorkloadWrapper{Workload: *baseWorkload.DeepCopy()}).
+				Admitted(false).
+				Obj(),
+			checks:  []kueue.AdmissionCheck{*baseCheck.DeepCopy()},
+			flavors: []kueue.ResourceFlavor{*baseFlavor1.DeepCopy(), *baseFlavor2.DeepCopy()},
+			configs: []kueue.ProvisioningRequestConfig{*provReqConfigWithRetryLimit(baseConfig, 1)},
+			requests: []autoscaling.ProvisioningRequest{
+				*requestWithConditions(baseRequest,
+					[]metav1.Condition{
+						{
+							Type:   autoscaling.Failed,
+							Status: metav1.ConditionFalse,
+						},
+						{
+							Type:   autoscaling.Provisioned,
+							Status: metav1.ConditionTrue,
+						},
+						{
+							Type:   autoscaling.Accepted,
+							Status: metav1.ConditionTrue,
+						},
+						{
+							Type:               autoscaling.BookingExpired,
+							Status:             metav1.ConditionTrue,
+							LastTransitionTime: metav1.NewTime(time.Now().Add(time.Hour * (-1))),
```

Comment:
This seems like having a potential to flake if we are on the edge of an hour. Can we use fake clock for that?

Reply:
Removed this as it seemed we could implement those test without it